### PR TITLE
Added new az-dcap-client ansible task and updated SGX1FLC getting started docs for devs.

### PR DESCRIPTION
**Pull request by johnkord**
This PR renames one of the ansible playbooks from "deploy_vanilla" to "deploy_for_oe_devs_acc". It also creates another ansible playbook for devs deploying to non-ACC VMs, which is identical to the ACC version with one difference: it doesn't install the az-dcap-client package.

To do this, this PR extracts the logic for installing the az-dcap-client package into its own ansible task.

This PR also removes the SGX1FLC (USE_LIBSGX) sections of the prereqs folder (The SGX1 non-FLC parts of prereqs has been kept around for now), and updates the SGX1FLCGettingStarted document accordingly.

### CODE REVIEW

File: docs/GettingStartedDocs/SGX1FLCGettingStarted.md [hunk 1]
```diff
@@ -17,45 +17,38 @@ This creates a source tree under the directory called openenclave.
...
 cd openenclave
-ansible-playbook scripts/ansible/ansible-include_task.yml --extra-vars "target=localhost included_task=tasks/ansible-install-prereqs.yml"
 ```
 
-If you are running in an Azure Confidential Compute VM and would like to use the attestation features, you should also run the following command from the root of the source tree:
-
+Ansible is required to install the project prerequisites. If not already installed, you can install it by running: 
 ```bash
-sudo make -C prereqs/az-dcap-client install
+scripts/ansible/install-ansible.sh
```
Comment (andschwa) [inline docs/GettingStartedDocs/SGX1FLCGettingStarted.md]: Maybe note that this needs to be run with `sudo` (or under root).

File: docs/GettingStartedDocs/SGX1FLCGettingStarted.md [hunk 2]
```diff
@@ -17,45 +17,38 @@ This creates a source tree under the directory called openenclave.
...
 
-## Install Intel SGX FLC support software packages
-
-There are two Intel packages needed for SGX1-FLC:
+If you are running in an Azure Confidential Compute (ACC) VM and would like to use the attestation features, you should also run the following command from the root of the source tree:
 
-- Intel(R) SGX driver with FLC support
-- Intel(R) SGX DCAP SDK
+```bash
+ansible-playbook scripts/ansible/deploy_for_oe_devs_acc.yml
```
Comment (andschwa) [inline docs/GettingStartedDocs/SGX1FLCGettingStarted.md]: Ditto.

**Comment by andschwa** [comment]
@johnkord Looking at the error I got, that's coming from the Intel packages, but they have 18.04 (Bionic) packages available here: https://download.01.org/intel-sgx/sgx_repo/ubuntu/dists/

Changing the Ansible scripts from xenial to bionic got it a lot further, until this one bailed out:

```
"The following packages have unmet dependencies:", " az-dcap-client : Depends: libcurl3 (>= 7.16.2) but it is not going to be installed"
```

With the following diff everything at least succeeded on 18.04:

```
diff --git a/scripts/ansible/deploy_for_oe_devs_acc.yml b/scripts/ansible/deploy_for_oe_devs_acc.yml
index 2c9eddffd..584335391 100644
--- a/scripts/ansible/deploy_for_oe_devs_acc.yml
+++ b/scripts/ansible/deploy_for_oe_devs_acc.yml
@@ -8,5 +8,5 @@
      - include_tasks: tasks/ansible-install-openenclave-deps.yml
      - include_tasks: tasks/ansible-install-sgx-driver.yml
      - include_tasks: tasks/ansible-install-sgx-packages.yml
-     - include_tasks: tasks/ansible-install-openenclave.yml
+     # - include_tasks: tasks/ansible-install-openenclave.yml
      - include_tasks: tasks/ansible-install-az-dcap-client.yml
diff --git a/scripts/ansible/tasks/ansible-install-az-dcap-client.yml b/scripts/ansible/tasks/ansible-install-az-dcap-client.yml
index eb912a5a2..7e9b3a85a 100644
--- a/scripts/ansible/tasks/ansible-install-az-dcap-client.yml
+++ b/scripts/ansible/tasks/ansible-install-az-dcap-client.yml
@@ -9,7 +9,7 @@
 - name: Add Microsoft APT repositories
   apt_repository: repo={{item}} state=present update_cache=yes
   with_items:
-    - deb [arch=amd64] https://packages.microsoft.com/ubuntu/16.04/prod xenial main
+    - deb [arch=amd64] https://packages.microsoft.com/ubuntu/18.04/prod bionic main

 - name: Install az-dcap-client
   apt:
diff --git a/scripts/ansible/tasks/ansible-install-openenclave-deps.yml b/scripts/ansible/tasks/ansible-install-openenclave-deps.yml
index 694e46cc7..e40a8caa2 100644
--- a/scripts/ansible/tasks/ansible-install-openenclave-deps.yml
+++ b/scripts/ansible/tasks/ansible-install-openenclave-deps.yml
@@ -10,8 +10,7 @@
 - name: Add LLVM and Microsoft APT repositories
   apt_repository: repo={{item}} state=present update_cache=yes
   with_items:
-    - deb http://apt.llvm.org/xenial/ llvm-toolchain-xenial-7 main
-    - deb [arch=amd64] https://packages.microsoft.com/ubuntu/16.04/prod xenial main
+    - deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-7 main

 - name: Install packages
   apt:
diff --git a/scripts/ansible/tasks/ansible-install-prereqs.yml b/scripts/ansible/tasks/ansible-install-prereqs.yml
index 4369d9e2a..a85293ebd 100644
--- a/scripts/ansible/tasks/ansible-install-prereqs.yml
+++ b/scripts/ansible/tasks/ansible-install-prereqs.yml
@@ -5,7 +5,7 @@
   apt_key: url="https://apt.llvm.org/llvm-snapshot.gpg.key" state=present

 - name: Add LLVM repo for CLang 7 with Spectre mitigation
-  apt_repository: repo="deb http://apt.llvm.org/xenial/ llvm-toolchain-xenial-7 main" state=present update_cache=yes
+  apt_repository: repo="deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-7 main" state=present update_cache=yes

 - name: Install Clang 7 to build with Spectre mitigation
   apt:
diff --git a/scripts/ansible/tasks/ansible-install-sgx-packages.yml b/scripts/ansible/tasks/ansible-install-sgx-packages.yml
index 7ddb9ddfb..a96a8853b 100644
--- a/scripts/ansible/tasks/ansible-install-sgx-packages.yml
+++ b/scripts/ansible/tasks/ansible-install-sgx-packages.yml
@@ -9,7 +9,7 @@
 - name: Add 01.org repository
   apt_repository: repo={{item}} state=present update_cache=yes
   with_items:
-    - deb [arch=amd64] https://download.01.org/intel-sgx/sgx_repo/ubuntu xenial main
+    - deb [arch=amd64] https://download.01.org/intel-sgx/sgx_repo/ubuntu bionic main

 - name: Install libsgx packages
   apt:
```

**Comment by andschwa** [comment]
Sorry @johnkord I am digging more into this. Why are we installing the `open-enclave` package for OE devs? Or is this guide for devs on top of OE? I read "OE Devs" as "developers _of_ OE".

**Comment by oprinmarius** [comment]
@andschwa   I currently have 2 PR's open adding more support for Ubuntu 18.04 .   The problem is that gcc tests are currently failing on 18.04. 

We need to add switch cases to ansible playbooks to support both versions:
ex: https://github.com/Microsoft/openenclave/blob/89a024feea1789316ae035aa8681b9f930d3699e/scripts/ansible/tasks/ansible-install-prereqs.yml
https://github.com/Microsoft/openenclave/blob/89a024feea1789316ae035aa8681b9f930d3699e/scripts/ansible/tasks/ansible-install-docker-packages.yml

**Comment by oprinmarius** [comment]
I created a new PR with all the ansible changes required for Ubuntu 18.04 : https://github.com/Microsoft/openenclave/pull/1334

**Comment by johnkord** [comment]
@andschwa As for 18.04 support, that's in progress, as @oprinmarius mentioned!

As for the phrase "OE Devs", that's true it's for developers of OE. If a user wants to develop an application that uses OE, they follow a different guide.

It's not necessary to install the `open-enclave` package. I could remove it if you think that's best! I simply included it as part of the .yml file because as an OE developer I try to make sure that both the OE source tests and package tests all work when I upgrade system tools or test new Intel package/driver releases.

**Comment by pushkarcMS** [comment]
Couple of things.  I am not sure how folks interpret "OE Devs".  What do we call application developers using OE for development?  Should we disambiguate "OE Contributors" from "OE Devlopers".  The ansible playbook for "non-ACC VMs", who is it meant for?  Currently we have no default dcap client.  Is this just preloading the work for when we will have a default dcap client from Intel?

**Comment by johnkord** [comment]
@pushkarcMS Yeah, I agree with you and think the term "OE Devs" is a little ambiguous. It might be better to rename this file to be "oe_contributors" instead of "oe_devs".

As for the playbook for the non-ACC VMs, I created it only to support a scenario that was specified in the documentation! I'm totally fine with leaving it out for now until a default dcap client exists (as it might be confusing).  What do you think would be best?

**Comment by johnkord** [comment]
@andschwa I think you're probably right. I'm going to remove it! Thanks Andy :)

**Comment by johnkord** [comment]
I'll leave this PR around for a day in case if anyone else has any requests, and then I'll merge if there are no objections

**Pull request merged**